assets/js/base/components/formatted-monetary-amount/index.tsx

```diff
@@ -17,6 +17,8 @@ import './style.scss';
 interface FormattedMonetaryAmountProps {
 	className?: string;
 	displayType?: NumberFormatProps[ 'displayType' ];
+	allowNegative?: boolean;
+	isAllowed?: ( formattedValue: NumberFormatValues ) => boolean;
```

Comment:
isAllowed is rather confusing, what are we supposed to allow here? what I understood from the PR is that it doesn’t allow negative numbers, for me `allowNegative` should be sufficient.

Also, do we need to pass a prop to `FormattedMonetaryAmount` just to support negative numbers? shouldn't just support them by default from what we pass to `value`?

Reply:
>isAllowed is rather confusing, what are we supposed to allow here? what I understood from the PR is that it doesn’t allow negative numbers, for me allowNegative should be sufficient.

I used `isAllowed` because with the function `withMaxValueLimit` and `withMinValueLimit` we can validate the input when the user changes the value in the input. About the naming, I chose `isAllowed` to be consistent with the props of the library `react-number-format`.

> Also, do we need to pass a prop to FormattedMonetaryAmount just to support negative numbers? shouldn't just support them by default from what we pass to value?

Do you mean that we should implement some custom logic to check if the value is negative? I preferred to use a ready solution that `react-number-format` give

Reply:
I still don't understand the goal of `isAllowed`, can't `allowNegative` do that instead? My comment is mostly why we need two.

Reply:
The goal `isAllowed` is to do an extra check on the input field:
If on the right input the user write 30, on the left input the user can't write 31

Reply:
It took me sometime to understand `isAllowed`. I initially thought this is a prop you introduced, I see it's part of `react-format-money`, Not a great naming from their side really, but I see its value now.